tests/code_style.py

```diff
@@ -0,0 +1,247 @@
+    """Validate if the test file exists in the expected location."""
+    test_location = _get_test_location(transformer)
+    if test_location is None:
+        return False
+
+    return test_location.exists()
+
+
+def _load_module_from_path(path):
+    """Return the module from a given ``PosixPath``."""
+    module_path = path.parent
+    module_name = path.name.split('.')[0]
+    if module_path.name == 'transformers':
+        module_path = f'rdt.transformers.{module_name}'
+    elif module_path.parent.name == 'transformers':
+        module_path = f'rdt.transformers.{module_path.parent.name}.{module_name}'
+    elif module_path.parent.name == 'addons':
+        module_path = f'rdt.transformers.addons.{module_path.parent.name}.{module_name}'
+
+    spec = importlib.util.spec_from_file_location(module_path, path)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+
+    return module
+
+
+def validate_test_names(transformer):
+    """Validate if the test methods are properly specified."""
+    test_file = _get_test_location(transformer)
+
+    # load the module
+    try:
+        module = _load_module_from_path(test_file)
+    except FileNotFoundError:
+        return False
+
+    # get test class
+    try:
+        test_class = getattr(module, f'Test{transformer.__name__}')
+
+    except AttributeError:
+        return False  # The test class doesn't exist
+
+    # get the test methods that start with test
+    test_functions = inspect.getmembers(test_class, predicate=inspect.isfunction)
+    test_functions = [
+        test
+        for test, _ in test_functions
+        if test.startswith('test')
+    ]
+
+    if not test_functions:
+        return False  # No test functions found
+
+    transformer_functions = [
+        name
+        for name, function in transformer.__dict__.items()
+        if isinstance(function, (FunctionType, classmethod, staticmethod))
+    ]
+
+    test_functions_is_valid = []
+
+    for test in test_functions:
+        count = len(test_functions_is_valid)
+        for transformer_function in transformer_functions:
+            simple_test = f'test_{transformer_function}'
+            described_test = f'test_{transformer_function}_'
+            if test.startswith(described_test):
+                test_functions_is_valid.append(True)
+            elif test.startswith(simple_test):
+                test_functions_is_valid.append(True)
+
+        if count == len(test_functions_is_valid):
+            test_functions_is_valid.append(False)
+
+    return all(test_functions_is_valid)
+
+
+def get_all_transformers():
+    """Return all the transformers."""
+    return BaseTransformer.get_subclasses()
+
+
+def get_all_transformers_as_dict():
+    """Return all the transformers."""
+    return {
+        transformer.__name__: transformer
+        for transformer in BaseTransformer.get_subclasses()
+    }
+
+
+def validate_transformer(transformer):
```

Comment:
I think this should be renamed to something like `test_transformer_code_style` and parameterized using `rdt.transformers.TRANSFORMERS` as input (values are test cases, keys are test ids):

```python3
@pytest.mark.parametrize('transformer', TRANSFORMERS.values(), ids=TRANSFORMERS.keys())
def test_transformer_code_style(transformer):
```

Inside the function, collecting results is not necessary: we just need to call the functions, which should raise an `AssertionError` if something is wrong.

After this, a `pytest tests/code_style.py -v --disable-warnings --no-header` call can be added to the `lint` command.